### PR TITLE
Code Review Pull Request

### PR DESCRIPTION
**Pull request by ehsteve**

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Changelog
+=========
+
+0.1.0 (2017-03-24)
```
Comment (nabobalis) [inline CHANGELOG.rst:4]: Where is the rest of the changelog?

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -0,0 +1,7 @@
+============
+Contributing
+============
+
+Contributions are welcome, and they are greatly appreciated! Every
+little bit helps, and credit will always be given.
+
```
Comment (nabobalis) [inline CONTRIBUTING.rst:7]: Could link to some guides about setting up a python development environment?

File: docs/conf.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+#html_sidebars = {}
+
+# The name of an image file (relative to this directory) to place at the top
+# of the sidebar.
+html_logo = 'logo/roentgen_logo.jpg'
+
+# The name of an image file (within the static path) to use as favicon of the
+# docs.  This file should be a Windows icon file (.ico) being 16x16 or 32x32
+# pixels large.
+#html_favicon = ''
```
Comment (nabobalis) [inline docs/conf.py:87]: DId you mean to use the default astropy theme for the docs with their favicon?

File: roentgen/util/util.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        return None
+
+
+def get_density(material_str):
+    """Given a material name return the default density"""
+    if is_an_element(material_str):
+        ind = get_atomic_number(material_str)-1
+        density = roentgen.elements[ind]["density"]
+    else:
+        # not using loc because table indexing is not yet stable
```
Comment (nabobalis) [inline roentgen/util/util.py:90]: This still true?

File: roentgen/tests/setup_package.py [hunk 5]
```diff
@@ -0,0 +1,3 @@
+def get_package_data():
+    return {
+        _ASTROPY_PACKAGE_NAME_ + '.tests': ['coveragerc']}
```
Comment (nabobalis) [inline roentgen/tests/setup_package.py:3]: Not sure this is needed anymore.

File: roentgen/tests/coveragerc [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+   # Don't complain if tests don't hit assertions
+   raise AssertionError
+   raise NotImplementedError
+
+   # Don't complain about script hooks
+   def main\(.*\):
+
+   # Ignore branches that don't pertain to this version of Python
+   pragma: py{ignore_python_version}
```
Comment (nabobalis) [inline roentgen/tests/coveragerc:31]: This should be in the setup.cfg file.

File: roentgen/setup_package.py [hunk 7]
```diff
@@ -0,0 +1,3 @@
+def get_package_data():
+    return {'roentgen': ['data/*.json', 'data/*.csv', 'data/elements/*.csv',
+            'data/compounds_mixtures/*.csv']}
```
Comment (nabobalis) [inline roentgen/setup_package.py:3]: This can be in the setup.cfg file.

File: roentgen/__init__.py [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+                                        format='csv', fast_reader=False))
+
+#emission_energies_file = os.path.join(_data_directory, 'emission_energies.csv')
+#emission_energies = QTable(ascii.read(emission_energies_file, fill_values=('', '-1')))
+#for colname in emission_energies.colnames[1:]:
+#    emission_energies[colname].unit = u.eV
+#emission_energies.meta = {"source": "Center for X-ray Optics and Advanced Light Source, X-Ray Data Booklet Table 1-2",
+#                          "publication date": "2009 October",
+#                          "url": "https://xdb.lbl.gov/Section1/Sec_1-2.html"}
+#emission_energies.add_index('z')
```
Comment (nabobalis) [inline roentgen/__init__.py:79]: What happened tho this?

File: gui/main.py [hunk 9]
```diff
@@ -0,0 +1,362 @@
...
+                                  DEFAULT_ENERGY_RESOLUTION), "keV")
+    y = response.response(energy)
+
+    #plot_title += f"{this_material.name} {this_material.thickness}"
+    #p.text += f'material {this_material_str} not recognized'
+
+    #plot_title += f" {air.name} {air.density.to('g / cm**3'):.2E} {air.thickness:.2f}"
+
+    #plot_title += f" {this_detector.name} {this_detector.thickness:.2f}"
+    #p.text += f'detector {this_detector_material_str} not recognized'
```
Comment (nabobalis) [inline gui/main.py:230]: What happened to this?

File: gui/main.py [hunk 10]
```diff
@@ -0,0 +1,362 @@
...
+ylog_checkbox = CheckboxGroup(labels=["y-log"], active=[0])
+
+# Set plot widgets
+#energy_low = Slider(title="energy (low) [keV]", value=DEFAULT_ENERGY_LOW,
+#                    start=1, end=5.0, step=0.1)
+#energy_high = Slider(title="energy (high) [keV]", value=DEFAULT_ENERGY_HIGH,
+#                        start=10, end=100, step=1)
+#energy_step = Slider(title="energy resolution [keV]",
+#                        value=DEFAULT_ENERGY_RESOLUTION, start=0.1, end=2,
+#                        step=0.1)
```
Comment (nabobalis) [inline gui/main.py:117]: Don't like widgets?

File: roentgen/absorption/material.py [hunk 11]
```diff
@@ -0,0 +1,281 @@
...
+        if (type(detector) is Material) or (detector is None):
+            self.detector = detector
+        else:
+            raise TypeError('Detector must be a Material or None')
+
+    def __repr__(self):
+        """Returns a human-readable representation."""
+        txt = "<Response optical path="
+        for material in self.optical_path:
+            txt += str(material)
```
Comment (dpshelio) [inline roentgen/absorption/material.py:211]:
```suggestion
        txt += ''.join(map(str, self.optical_path))
```

File: roentgen/absorption/material.py [hunk 12]
```diff
@@ -0,0 +1,281 @@
...
+            transmission *= this_transmission
+        if self.detector is None:
+            detector_absorption = np.ones(len(energy), dtype=np.float)
+        else:
+            detector_absorption = self.detector.absorption(energy)
+
+        return transmission * detector_absorption
+
+
+class MassAttenuationCoefficient(object):
```
Comment (dpshelio) [inline roentgen/absorption/material.py:233]:
```suggestion
class MassAttenuationCoefficient:
```

File: roentgen/absorption/material.py [hunk 13]
```diff
@@ -0,0 +1,281 @@
...
+            return Compound(self.materials + other.materials)
+        else:
+            raise ValueError("Can't add <Compound> and " + str(other))
+
+    def __repr__(self):
+        """Returns a human-readable representation."""
+        txt = "<Compound "
+        for material in self.materials:
+            txt += str(material)
+        return txt + ">"
```
Comment (dpshelio) [inline roentgen/absorption/material.py:141]: Similar to below (`Response`)

File: roentgen/absorption/material.py [hunk 14]
```diff
@@ -0,0 +1,281 @@
...
+        self.is_detector = is_detector
+
+        if density is None:
+            self.density = get_density(material_str)
+        else:
+            self.density = density
+
+    def __repr__(self):
+        """Returns a human-readable representation."""
+        txt = "<Material " + str(self.name) + " (" + str(self.name) + ") "
```
Comment (dpshelio) [inline roentgen/absorption/material.py:66]:
```suggestion
        txt = f"<Material {self.name} ({self.name}) "
```

File: roentgen/absorption/material.py [hunk 15]
```diff
@@ -0,0 +1,281 @@
...
+        self.name = material_str
+        self.thickness = thickness
+        self.mass_attenuation_coefficient = MassAttenuationCoefficient(material_str)
+        self.name = self.mass_attenuation_coefficient.name
+        self.is_detector = is_detector
+
+        if density is None:
+            self.density = get_density(material_str)
+        else:
+            self.density = density
```
Comment (dpshelio) [inline roentgen/absorption/material.py:62]: I find it more readable the other way:
```python
if density:
    ...
else:
    ...
```

File: roentgen/absorption/material.py [hunk 16]
```diff
@@ -0,0 +1,281 @@
...
+    is_detector : bool
+        A property to hold information on whether absorption or transmission
+        is important. Used for compounds.
+
+    Examples
+    --------
+    >>> from roentgen.absorption.material import Material
+    >>> import astropy.units as u
+    >>> detector = Material('cdte', 500 * u.um)
+    >>> thermal_blankets = Material('mylar', 0.5 * u.mm)
```
Comment (dpshelio) [inline roentgen/absorption/material.py:47]: Make it as doctest and run something that produces an output.

File: roentgen/absorption/material.py [hunk 17]
```diff
@@ -0,0 +1,281 @@
...
+"""
+
+from __future__ import absolute_import
+
+import numpy as np
+import os
+import astropy.units as u
+from scipy import interpolate
+import roentgen
+from roentgen.util import get_atomic_number, get_density, get_compound_index, get_element_symbol, is_an_element, is_in_known_compounds
```
Comment (dpshelio) [inline roentgen/absorption/material.py:11]:
```suggestion
from roentgen.util import (get_atomic_number, get_density, get_compound_index,
                           get_element_symbol, is_an_element, is_in_known_compounds)
```

File: roentgen/absorption/material.py [hunk 18]
```diff
@@ -0,0 +1,281 @@
+"""
+"""
+
+from __future__ import absolute_import
```
Comment (dpshelio) [inline roentgen/absorption/material.py:4]: Why?... are you planning to use python2? :scream_cat:

File: roentgen/absorption/material.py [hunk 19]
```diff
@@ -0,0 +1,281 @@
...
+
+import numpy as np
+import os
+import astropy.units as u
+from scipy import interpolate
```
Comment (dpshelio) [inline roentgen/absorption/material.py:9]:
```suggestion
import os

import astropy.units as u
import numpy as np
from scipy import interpolate

```

`isort` will do that for you.

File: licenses/TEMPLATE_LICENCE.rst [hunk 20]
```diff
@@ -0,0 +1,31 @@
+This project is based upon the Astropy package template
```
Comment (dpshelio) [inline licenses/TEMPLATE_LICENCE.rst:1]: I'm sure you want to change that `template` to something real...

File: gui/main.py [hunk 21]
```diff
@@ -0,0 +1,362 @@
...
+        air_thickness_input.disabled = True
+        air_temperature_input.disabled = True
+    if 2 in new:
+        detector_material_input.disabled = False
+        detector_thickness_input.disabled = False
+        detector_density_input.disabled = False
+    if 2 not in new:
+        detector_material_input.disabled = True
+        detector_thickness_input.disabled = True
+        detector_density_input.disabled = True
```
Comment (dpshelio) [inline gui/main.py:281]:
```suggestion
    new_2 = (2 in new)
    detector_material_input.disabled = not new_2
    detector_thickness_input.disabled = not new_2
    detector_density_input.disabled = not new_2
```

**Comment by ehsteve** [comment]
Thanks for the comments @dpshelio! I appreciate it.

File: docs/conf.py (hunk 3 above)
Comment (ehsteve) [reply docs/conf.py:87]: moving to sunpy theme

File: CONTRIBUTING.rst (hunk 2 above)
Comment (ehsteve) [reply CONTRIBUTING.rst:7]: added link to sunpy newcomer guide.

**Pull request closed without merging**